File: frontend/src/api/wiki.ts
import { BASE_URL as BASE, defaultHeaders as headers, handleResponse } from './client'

export interface WikiPageListItem {
  id: string
  slug: string
  titolo: string
  last_compiled_at: string | null
  compilation_model: string | null
  version: number
  source_raw_ids: string[] | null
}

export interface WikiPageRead extends WikiPageListItem {
  contenuto_md: string
  links: string[] | null
}

export interface WikiIndexRead {
  id: number
  tree_md: string | null
  last_rebuilt_at: string | null
}

export async function fetchWikiPages(): Promise<WikiPageListItem[]> {
  const res = await fetch(`${BASE}/wiki/pages`, { headers })
  return handleResponse(res)
}

export async function fetchWikiPage(slug: string): Promise<WikiPageRead> {
  const res = await fetch(`${BASE}/wiki/pages/${encodeURIComponent(slug)}`, { headers })
  return handleResponse(res)
}

export async function fetchWikiIndex(): Promise<WikiIndexRead> {
  const res = await fetch(`${BASE}/wiki/index`, { headers })
  return handleResponse(res)
}

File: backend/app/schemas/wiki.py
import uuid
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, ConfigDict


class WikiPageListItem(BaseModel):
    id: uuid.UUID
    slug: str
    titolo: str
    last_compiled_at: Optional[datetime]
    compilation_model: Optional[str]
    version: int
    source_raw_ids: Optional[list]

    model_config = ConfigDict(from_attributes=True)


class WikiPageRead(BaseModel):
    id: uuid.UUID
    slug: str
    titolo: str
    contenuto_md: str
    links: Optional[list]
    source_raw_ids: Optional[list]
    last_compiled_at: Optional[datetime]
    compilation_model: Optional[str]
    version: int

    model_config = ConfigDict(from_attributes=True)


class WikiIndexRead(BaseModel):
    id: int
    tree_md: Optional[str]
    last_rebuilt_at: Optional[datetime]

    model_config = ConfigDict(from_attributes=True)


class CompilationLogRead(BaseModel):
    id: uuid.UUID
    triggered_at: datetime
    trigger_type: str
    affected_raw_ids: Optional[list]
    affected_wiki_slugs: Optional[list]
    duration_ms: Optional[int]
    tokens_used: Optional[int]
    model_used: Optional[str]
    status: str
    error: Optional[str]

    model_config = ConfigDict(from_attributes=True)


class CompilationLogListResponse(BaseModel):
    items: list[CompilationLogRead]
    total: int
    page: int
    page_size: int
